dev/TreeView/TreeView.idl

```diff
@@ -124,6 +124,7 @@ unsealed runtimeclass TreeView : Windows.UI.Xaml.Controls.Control
     event Windows.Foundation.TypedEventHandler<TreeView, TreeViewItemInvokedEventArgs> ItemInvoked;
     event Windows.Foundation.TypedEventHandler<TreeView, TreeViewExpandingEventArgs> Expanding;
     event Windows.Foundation.TypedEventHandler<TreeView, TreeViewCollapsedEventArgs> Collapsed;
+    event Windows.Foundation.TypedEventHandler<TreeView, Windows.UI.Xaml.Controls.SelectionChangedEventArgs> SelectionChanged;
```

Comment:
We likely want to use a new type here instead of exposing the one from ListView directly.

Reply:
Happy to intoduce a new type if needed. Is there any guidance regarding re-using the existing `SelectionChangedEventArgs` type? For example, the `RadioButtons` control is reusing it for the `SelectionChanged` event.

I guess it also depends what we want to expose in the event args for the `TreeView.SelectionChanged` event. I thought what is `SelectionChangedEventArgs` would make perfect sense for the `TreeView` control:

```csharp
public class SelectionChangedEventArgs : RoutedEventArgs, ISelectionChangedEventArgs
{
   public IList<object> AddedItems { get; } 
   public IList<object> RemovedItems { get; }
 }
```

Reply:
In general sharing event args between types is a little risky, it means that if you want to change how selection is reported for TreeView, you'd have to also update how it is reported for list view since they use the same type...  It was probably a mistake for RadioButtons to use this type.

Reply:
Now using a custom type for the event args.

Reply:
Can you please add the preview tag to this addition? We cannot publish it without API review but you shouldn't be blocked on that.

---
In reply to: [389258858](https://github.com/microsoft/microsoft-ui-xaml/pull/2063#discussion_r389258858) [](ancestors = 389258858)

Reply:
Preview tag added. @YuliKl how can we make progress on the API proposal?

Reply:
@StephenLPeters what is the next step regarding the API definition? Do we want to merge this "as is" with the preview tag? Do we need to pitch the API to your team?

Reply:
I think the default behavior is to merge with preview tag and create a PR over at https://github.com/microsoft/microsoft-ui-xaml-specs/ for the API spec.

If you want, I can help you with the latter thing.

Reply:
We have a number of API's which are currently in the preview state, our team has a process for reviewing these API's before we commit to supporting them long term. I think @ranjeshj and @predavid would have some input here. Generally the api spec that @chingucoding mentioned would be required.

Reply:
@chingucoding I don't see a `TreeView` folder in the API spec repo. Should I create one and just mention the change we're talking about in this PR (adding the `SelectionChanged` event + the event args type)?

Reply:
In that case, you can create a new folder under "active" for TreeView and put the new spec file in there. In the repository, there is a template file that you can use, where you can see what you should write in the spec. 
In our case, that would be the event and the event types.

Reply:
@predavid as FYI for the API spec. @japf Thanks for making this change!